### ISSUE TITLE
AJAXify Formspree

### ISSUE THREAD

**Issue by Joshfindit** [johnotander/pixyll#204]
In case you are interested, here is some code that I just worked out (and tested):

```
<form>
  <input name="name" type='text' placeholder='Your name (optional)'>
  <input name="email" type="hidden" placeholder='Your email'>
  <textarea type="text" name="message" placeholder='Your message (optional)'></textarea>
  <button type='submit' class="button button-blue button-big mobile-block">Send message</button>
</form>

<script>
	var contactForm = document.querySelector('form'),
    inputName = contactForm.querySelector('[name="name"]'),
    inputEmail = contactForm.querySelector('[name="email"]'),
    textAreaMessage = contactForm.querySelector('[name="message"]'),
    sendButton = contactForm.querySelector('button');

    sendButton.addEventListener('click', function(event){
      event.preventDefault(); // prevent the form to do the post.

      sendButton.innerHTML = 'sending..';

      var xhr = new XMLHttpRequest();
      xhr.open('POST', '//formspree.io/yourmail@example.com', true);
      xhr.setRequestHeader("Accept", "application/json")
      xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded")

      xhr.send(
        "name=" + inputName.value +
        "&email=" + inputEmail.value +
        "&message=" + textAreaMessage.value);

      xhr.onloadend = function (res) {
        if (res.target.status === 200){
          sendButton.innerHTML = 'Message sent!';
        }
        else {
          sendButton.innerHTML = 'Error!';
        }
      }
    });
</script>
```
I can do a PR if @johnotander chooses whether to inline the JS (and where), or include it as a separate file.

**Comment by johnotander** [comment]
Awesome! I'd love to merge a PR for this if it were configurable via `_config.yml` (defaulting to the current behavior). That way if `ajax_contact_form` (or some better named config key) is set to true it will `<%= include 'ajax_form_js' %>` which contains the inlined JS.

**Comment by Joshfindit** [comment]
PR created: https://github.com/johnotander/pixyll/pull/205

**Comment by gregist** [comment]
@johnotander `{% include ajaxify_content_form.js %}` and filename `ajaxify_content_form.html` are inconsistent. please change either.

**Comment by Joshfindit** [comment]
Addressed in PR https://github.com/johnotander/pixyll/pull/206.
Apologies; my mistake.

**Comment by johnotander** [comment]
Closed via #205 and #206
